core/sql/ustat/hs_globals.cpp

```diff
@@ -2887,6 +2890,22 @@ Lng32 HSGlobalsClass::Initialize()
         HSHandleError(retcode);
         TM->Commit(); // Must commit this transaction (even if schema didn't get created).
       }
+                                              /*=====================================*/
+                                              /*   CREATE HBASE STATS SCHEMA         */
+                                              /*   typically as trafodion.hbasestats */
+                                              /*=====================================*/
+    if (isNativeHbaseCat(objDef->getCatName()))
+      {
+        HSTranMan *TM = HSTranMan::Instance(); // Must have transaction around this.
+        TM->Begin("Create schema for native hbase stats.");
+        NAString ddl = "CREATE SCHEMA IF NOT EXISTS ";
+        ddl.append(HBASE_STATS_CATALOG).append('.').append(HBASE_STATS_SCHEMA);
```

Comment:
I will discuss this with you.  Sounds like we need to create two schemas
(one for native HBase and one for  native HIVE) during the cluster setup
step.  During update stats, it is too late because a regular user can
perform the update stats.

On Mon, Sep 21, 2015 at 1:21 PM, Roberta Marton <notifications@github.com>
wrote:

> In core/sql/ustat/hs_globals.cpp
> <https://github.com/apache/incubator-trafodion/pull/87#discussion_r40006091>
> :
>
> > @@ -2887,6 +2890,22 @@ Lng32 HSGlobalsClass::Initialize()
> >          HSHandleError(retcode);
> >          TM->Commit(); // Must commit this transaction (even if schema didn't get created).
> >        }
> > +                                              /*=====================================*/
> > +                                              /*   CREATE HBASE STATS SCHEMA         */
> > +                                              /*   typically as trafodion.hbasestats */
> > +                                              /*=====================================*/
> > +    if (isNativeHbaseCat(objDef->getCatName()))
> > +      {
> > +        HSTranMan *TM = HSTranMan::Instance(); // Must have transaction around this.
> > +        TM->Begin("Create schema for native hbase stats.");
> > +        NAString ddl = "CREATE SCHEMA IF NOT EXISTS ";
> > +        ddl.append(HBASE_STATS_CATALOG).append('.').append(HBASE_STATS_SCHEMA);
>
> Do you want these schemas to be private or shared? All tables in a private
> schema are owned by the creator. Tables in a shared schema are owned by the
> creator.
> Who owns these schemas? You should add a authorization clause to specify
> the owner. You could make DB__ROOT the owner or add a new role as the
> owner.
> The MANAGE_STATISTICS is a privilege we support today. Anyone with the
> MANAGE_STATISTICS privilege can perform update statistics. Will this work
> today with sample tables.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/87/files#r40006091>.
>



-- 
Regards, --Qifan